File: level2/스킬트리/Solution.java
import java.util.*;

class Solution {
    public int solution(String skill, String[] skill_trees) {
        int answer = 0;
        for (int i = 0; i < skill_trees.length; i++) {
            if (isSkillTree(skill, skill_trees[i]))
                answer++;
        }
        return answer;
    }

    private boolean isSkillTree(String skill, String skill_tree) {
        return skill.startsWith(skill_tree.replaceAll("[^" + skill + "]", ""));
    }

    public static void main(String[] args) {
        Solution s = new Solution();
        System.out.println(s.solution("CBD", new String[] {"BACDE", "CBADF", "AECB", "BDA"}));
    }
}

File: level2/스킬트리/solution.rb
def solution(skill, skill_trees)
  answer = 0
  skill_trees.each { |st| answer += 1 if skill.start_with? st.gsub(Regexp.new("[^#{skill}]"), "") }
  return answer
end

p solution "CBD", ["BACDE", "CBADF", "AECB", "BDA"]
